src/components/Planning/GroupDialog/GroupDialog.tsx

```diff
@@ -0,0 +1,292 @@
+import { useState, MouseEvent, FC, useCallback } from 'react';
+import { Button } from 'hds-react/components/Button';
+import { Dialog } from 'hds-react/components/Dialog';
+import { useTranslation } from 'react-i18next';
+import './styles.css';
+
+import { TextField, SelectField } from '@/components/shared';
+import { IOption } from '@/interfaces/common';
+import { IconAngleUp, IconAngleDown } from 'hds-react/icons';
+import GroupProjectSearch from './GroupProjectSearch';
+import useGroupForm from '@/forms/useGroupForm';
+import { useAppDispatch } from '@/hooks/common';
+import { IGroupForm } from '@/interfaces/formInterfaces';
+import { postGroupThunk } from '@/reducers/groupSlice';
+import { IGroupRequest } from '@/interfaces/groupInterfaces';
+
+interface IDialogProps {
+  handleClose: () => void;
+  isOpen: boolean;
+}
+
+interface IFormState {
+  showAdvanceFields: boolean;
+}
+interface IDialogButtonState {
+  isOpen: boolean;
+}
+const buildRequestPayload = (form: IGroupForm): IGroupRequest => {
+  // submit Class or subclass if present, submit division or subDivision if present, submit a name, submit projects
+  const payload: IGroupRequest = {
+    name: '',
+    classRelation: '',
+    districtRelation: '',
+    projects: [],
+  };
+
+  payload.name = form.name;
+  payload.classRelation = form.subClass?.value || form.class?.value || '';
+  payload.districtRelation = form.subDivision?.value || form.division?.value || '';
+  payload.projects =
+    form.projectsForSubmit.length > 0 ? form.projectsForSubmit.map((p) => p.value) : [];
+
+  return payload;
```

Comment:
Could this function just do this?

`return {
    name: form.name,
    classRelation: form.subClass?.value || form.class?.value || '',
    districtRelation:form.subDivision?.value || form.division?.value || '',
    projects: form.projectsForSubmit.map((p) => p.value),
  };`